datastream-utils/src/main/java/com/linkedin/datastream/common/LogUtils.java

```diff
@@ -92,4 +95,68 @@ public static String logSummarizedTopicPartitionsMapping(List<String> partitions
             .append(logNumberArrayInRange(topicPartitionsMap.get(topicName))))
         .collect(Collectors.joining(", "));
   }
+
+  /**
+   * prints one log line for each string smaller than a given size limit, splitting longer messages into multiple part
+   * @param log logger to use for logging
+   * @param message string to be logged
+   * @param contextPrefix string that provides context for what is being logged, which will be added right before the
+   *                      actual message being logged (e.g. contextPrefix="Live task: ", Log line="Live task: message")
+   * @param sizeLimit size limit of each log line
+   */
+  public static void logStringsUnderSizeLimit(Logger log, String message, String contextPrefix, double sizeLimit) {
+    if (sizeLimit <= BUFFER_1KB) {
+      throw new IllegalArgumentException("Log size limit cannot be set to less than or equal to 1KB");
+    } else if (isLessThanSizeLimit(contextPrefix, BUFFER_1KB)) {
+      throw new IllegalArgumentException("Context prefix cannot be longer than 1KB in size");
+    } else {
+      int bufferAdjustedSizeLimit = (int) (sizeLimit - BUFFER_1KB);
+      logStringsUnderSizeLimit(log, message, contextPrefix, 1, sizeLimit, bufferAdjustedSizeLimit);
+    }
+  }
+
+  /**
+   * helper function to print and keep log lines under a certain size limit
+   * @param log logger to use for logging
+   * @param message string to be logged
+   * @param contextPrefix string that provides context for what is being logged
+   * @param part printed in the log message to keep track of how many parts a larger message has been split into
+   * @param sizeLimit size limit of each log line
+   * @param adjustedSizeLimit buffer adjusted size limit to account for extra bytes in the log line (e.g. timestamp, avro wrapping)
+   */
+  private static void logStringsUnderSizeLimit(Logger log, String message, String contextPrefix, int part,
+      double sizeLimit, int adjustedSizeLimit) {
+    if (isLessThanSizeLimit(message, sizeLimit)) {
```

Comment:
just pass the `adjustedSizeLimit` here to simplify the `isLessThanSizeLimit` function.